### ISSUE TITLE
Alternative method to initialise `Select`

### ISSUE THREAD

**Issue by rodrigogiraoserrao** [Textualize/textual#3691]
I propose we extend `Select.__init__` or create a classmethod `Select.from_values` to accept an iterable of `SelectType` and automatically build the labels by converting them to strings.

I find that often we just want labels that are string representations of the values we want to select:

```py
Select.from_values(data_source)
# vs
Select([(str(datum), datum) for datum in data_source])
```

`data_source` is often just a list of strings or paths and the comprehension to create the tuple looks really awkward.

For example, in the `Select` docs we do this:

```py
# ...
yield Select((line, line) for line in LINES)
```

I personally don't like it at all.

**Comment by azinneck0485** [comment]
Can I work on this?

I have done a quick prototype solution following the class method approach you suggested.  Before I go forward with formalizing the prototype, I wanted to ask: is this preferrable to overloading the constructor?

**Comment by rodrigogiraoserrao** [comment]
Hey @azinneck0485 thanks for your interest!
I don't see any particular reason why you wouldn't be able to work on this. 🤷 

Let us just make sure that the class method is a good idea and then we'll let you know.
If you're working on this, make sure to read the [contributing guidelines](https://github.com/Textualize/textual/blob/main/CONTRIBUTING.md) for an experience as smooth as possible!

**Comment by willmcgugan** [comment]
I think we should go ahead and add the classmethod. @azinneck0485 are you happy to contribute that?

**Comment by azinneck0485** [comment]
Yes, I will contribute that change and get a PR in once I've tested it.

**Comment by rodrigogiraoserrao** [comment]
Excellent @azinneck0485!
I've assigned this to you. (I've also assigned it to me for bookkeeping purposes.)
Take your time and then make sure to ping me once your PR is in!

**Comment by azinneck0485** [comment]
Hi @rodrigogiraoserrao -- I have made the code changes and done some testing on my own.  I have also updated the documentation by adding an example (which is just the existing example modified to use the new class method to create the select control).  All changes have been pushed to a branch on my fork.

However, I am not yet ready for a PR as I have run into some problems running `pytest`.  Although I have only added 3 tests (replicating the existing tests for Select()), 16 tests are failing in total.  13 of those failing tests are from `test_text_area_language_rendering` and `test_text_area_themes` and are due to missing syntax highlighting in the tests I have run.

I don't want to accidentally update snapshots erroneously, so I wanted to check and see if you had any tips on what I should look into to ensure these tests pass?

**Comment by rodrigogiraoserrao** [comment]
Hi @azinneck0485, thanks for the update!
If you're running Python 3.7, you won't be able to install the dependencies `tree-sitter` and `tree_sitter_languages`, which require Python 3.8 or greater.
If you're running Python 3.8 or greater, make sure you have the dependencies installed.

In my case, I'm running Python 3.7 and if I run `make test`, this is what the final output looks like:

![Screenshot 2023-11-23 at 11 01 48](https://github.com/Textualize/textual/assets/5621605/a70018e7-11d8-486d-8246-4d3e750c463d)

Notice how there's a red warning saying there are 13 unused snapshots, which makes it look like in the end the tests failed (see the Error 1 at the bottom).

However, at the same time, pytest reports that those tests were skipped.

In short, there's an issue with the snapshot tests reporting (which is a problem in the tool we use for the snapshots) and it's ok for those 13 snapshots to report that “error”.

In order to update _only_ the snapshot(s) you created, first make sure they look ok.
(I.e., run the test, open the snapshot report error, and make sure the snapshot looks like it is intended to look.)
Then, run the test again with the flag `--snapshot-update`, but _only_ on your tests.

For example, if you had a snapshot test called `test_azinneck_snap`, you could run `pytest -vv tests/snapshot_tests/test_snapshots.py::test_azinneck_snap --snapshot-update` and it would update only your snapshot.

**Comment by azinneck0485** [comment]
@rodrigogiraoserrao, thanks for you tips!  After adding those packages, the failures went away and I was able to successfully run the new snapshot tests I created.

I have added PR #3743 that includes the code change, test changes, and documentation changes.  Please let me know if there are additional changes that I need to make.

Thanks again for all your help.
